Controllers/OperationController.cs

```diff
@@ -146,5 +152,18 @@ public async Task<HttpResponseMessage> Invoke(OperationInfo info)
 
             return await _armRepository.InvokeAsync(Request, executeRequest);
         }
+
+        private static bool IsValidHost()
+        {
+            if (HttpContext.Current.Request.Url.IsLoopback)
+            {
+                return true;
+            }
+
+            // For Azure scenarios we do extra checks for cross domains
+            var requestHost = HttpContext.Current.Request.UrlReferrer?.Host;
```

Comment:
check Referrer, if not exists then Origin.   Certain browser has strict Refer policy and will only pass origin.